### PR TITLE
Fix false positive ``unused-private-member`` for accessing attributes in a class using ``cls``

### LINKED ISSUE #4681: False-positive `unused-private-member` with class name
### Steps to reproduce

```py
class Singleton:
    __instance = None

    @staticmethod
    def instance():
        if Singleton.__instance is None:
            Singleton()
        return Singleton.__instance

    def __init__(self):
        try:
            Singleton.__instance = 42  # Error here
        except Exception:  # pylint: disable=broad-except
            print("Error")
            Singleton.__instance = False  # Error here
```

**or**
```py
class Singleton:
    __instance = None

    @classmethod  # Use class method here
    def instance(cls):
        if cls.__instance is None:
            cls()
        return cls.__instance

    def __init__(self):
        try:
            Singleton.__instance = 42  # Error here
        except Exception:  # pylint: disable=broad-except
            print("Error")
            Singleton.__instance = False  # Error here
```

### Current behavior

```
test.py:12:12: W0238: Unused private member `Singleton.__instance` (unused-private-member)
test.py:15:12: W0238: Unused private member `Singleton.__instance` (unused-private-member)
```

### Expected behavior
No errors

### pylint --version output

```
pylint 2.9.4-dev0
astroid 2.6.3-dev0
Python 3.10.0b3
```

### Additional information
Probably needs to be fixed similar to #4663

### PR DESCRIPTION
**Pull request by yushao2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Fix false positive for ``unused-private-member`` when accessing attribs in class using ``cls`` by extending logic

Closes #4681

### CODE REVIEW

File: pylint/checkers/classes.py [hunk 1]
```diff
@@ -1000,11 +1000,15 @@ def _check_unused_private_attributes(self, node: nodes.ClassDef) -> None:
...
-                    "cls",
-                    "self",
-                ]:
-                    # If assigned to cls.attrib, can be accessed by cls/self
+                if (
+                    assign_attr.expr.name
+                    in [
+                        "cls",
+                        node.name,
+                    ]
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/classes.py:1008]: Why do we not have "self" in this list, and shouldn't we create a variable ``this_class_names =  ["cls", "self", node.name]``  ? Also, is this compatible with https://github.com/PyCQA/pylint/pull/4949#pullrequestreview-746579519 ?

**Comment by Pierre-Sassoulas** [comment]
@yushao2 is this ready for review ? We might want to do the refactor later so we can ship this in pylint 2.11.

File: pylint/checkers/classes.py (hunk 1 above)
Comment (yushao2) [reply pylint/checkers/classes.py:1011]: if assignment is done to attributes of `cls` or the class name itself, we can access it via `cls`, `self` and the class name.

however, if it's assigned to `self`, then it is only accessible by `self`

**Comment by yushao2** [comment]
> @yushao2 is this ready for review ? We might want to do the refactor later so we can ship this in pylint 2.11.

yep, please go ahead and let me know of any feedback :)

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**